app/assets/javascripts/analytics.js

```diff
@@ -0,0 +1,209 @@
+        return d.value;
+      })
+      .attr("x", (d, i) => {
+        return i * (this.width / this.data.length) + (this.width / this.data.length - this.barPadding) / 2;
+      })
+      .attr("y", this.setYForLabel.bind(this) )
+      .attr("class", "label")
+      .attr("text-anchor", "middle")
+      .attr("font-family", "sans-serif")
+      .attr("font-size", "11px")
+      .attr("fill", this.setFillForLabel.bind(this) );
+  }
+
+  drawAxis () {
+    var xScale = d3.scale.ordinal()
+                   .domain(this.data.map( (d, i) => {
+                     return moment(d.date).format(this.axisDateFormat);
+                   }))
+                   .rangeBands([0, this.width]);
+
+
+    var xAxis = d3.svg.axis()
+                  .scale(xScale)
+                  .orient('bottom')
+
+
+    this.svg.append("g")
+        .attr("class", "x axis")
+        .attr("transform", `translate(0, ${(this.height)})`)
+        .call(xAxis);
+  }
+
+  setYScale (dataset) {
+    this.scale = d3.scale.linear()
+                   .domain([0, d3.max( dataset, (d) => {
+                     return d.value;
+                   })])
+                   .range( [0, this.height]);
+
+  }
+}
+
+class Dashboard {
+  constructor (id, chart) {
+    this.id         = id;
+    this.chart      = chart;
+    this.$totalAll  = $('.total-actions-all');
+    this.$totalNew  = $('.total-actions-new');
+
+    $('button#refresh-data').on('click', this.refreshData.bind(this) );
+  }
+
+  getData (cb) {
+    d3.json(`/api/pages/${this.id}/analytics.json`, (json)  => {
+      if(cb) {
+        cb(json);
+        this.setCounters(json.totals);
+      }
+    } );
+  }
+
+  setCounters (totals) {
+    this.$totalAll.html(totals.all_total);
+    this.$totalNew.html(totals.new_total);
+  }
+
+  refreshData () {
+    this.getData( (data) => {
+      this.chart.data = data.hours;
+      this.chart.update();
+    } );
+  }
+
+}
+
+var createMiniChart = (className, data) => {
+  var svg = d3.select(`#analytics-dashboard .${className} .chart`)
+
+  var chart = new AnalyticsDashboard();
+  chart.width   = 360;
+  chart.height  = 70;
+  chart.fill    = 'rgba(51,51,51,0.3)';
+  chart.data    = data;
+  chart.xAxis   = false;
+  chart.labels  = false;
+  chart.svg     = svg;
+  return chart;
+}
+
+module.exports = {
+  makeDashboard (pageId) {
+    var shortChartSVG = d3.select("#analytics-dashboard .short-view .chart"),
+        chart         = new AnalyticsDashboard(),
+        d             = new Dashboard(pageId, chart);
+
+    d.getData( (data) => {
+      chart.width     = 650;
+      chart.height    = 280;
+      chart.data      = data.hours;
+      chart.fill      = 'rgba(51,51,51,1)';
+      chart.svg       = shortChartSVG;
+      chart.axisDateFormat = 'HH a';
+      chart.render();
+
+
+      createMiniChart('mini-total', data.days_total.reverse())
+        .render();
+
+      createMiniChart('mini-new',   data.days_new.reverse())
+        .render();
+    });
+  }
+}
+
```

Comment:
this code looks really nice. It's a great use of es6 classes and it's quite clear.